### DISCUSSION TITLE
toggle options -> xaxis -> type from "datetime" to "category" make xaxis labels show numbers

### DISCUSSION THREAD

**Discussion by ktalanDev** [apexcharts/apexcharts.js#4818] [Q&A]
I want to display a graph with two types of data on the x-axis.
One with strings and another with timestamp.
At initialization I have no problems but once **I switch from datetime to category the x axis displays numbers instead of strings**.
I tried 2 methods to update the options by going through a state and using the method "updateOptions".

code : https://codesandbox.io/p/devbox/4lgy3d

![good initialization](https://github.com/user-attachments/assets/185968db-c460-4d19-b47b-fb7d56d329b8)
![good switch to date](https://github.com/user-attachments/assets/38d08b5e-fa92-4149-9d9a-3d59df353d1a)
![wrong return to string x-axis](https://github.com/user-attachments/assets/c5a25593-4d7f-4e51-be98-ef54f7347c6c)
![series data string x-axis](https://github.com/user-attachments/assets/29320ea5-6d43-490b-b466-369102387ce5)
![series data date x-axis](https://github.com/user-attachments/assets/d48f3796-4a67-4aff-84b5-a6c257f48ee9)

**Comment by dilenio** [comment]
Also, when resizing the browser window, the chart inside its container becomes incorrect, shifting to the right
